Bug in patching of rcParams

I'm running into the following failure over in glue-jupyter:

```
/usr/lib/python3.11/contextlib.py:137: in __enter__
    return next(self.gen)
../../.tox/py311-test-visual/lib/python3.11/site-packages/pytest_mpl/plugin.py:318: in switch_backend
    prev_backend = matplotlib.get_backend().lower()
../../.tox/py311-test-visual/lib/python3.11/site-packages/matplotlib/__init__.py:1275: in get_backend
    return rcParams['backend']
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = RcParamsScoped({'_internal.classic_mode': True,
                'agg.path.chunksize': 0,
                'animation.bi...          'ytick.minor.visible': False,
                'ytick.minor.width': 0.5,
                'ytick.right': True})
key = 'backend'

    def __getitem__(self, key):
>       return self._get_context_dict().__getitem__(key)
E       KeyError: 'backend'
```

basically it seems the backend key is missing from the patched rcParams.

@maartenbreddels - here's a minimal reproducer:

```
python -m venv clean
source clean/bin/activate
pip install solara[pytest] "pytest-playwright<0.6" matplotlib pytest-mpl
```

then create a file ``test.py`` with:

```python
import pytest

@pytest.mark.mpl_image_compare()
def test_basic(
    tmp_path,
    page_session,
    solara_test,
):
    pass
```

then:

```
pytest test.py --mpl
```

The error is:

```
self = RcParamsScoped({'_internal.classic_mode': 'True',
                'agg.path.chunksize': '0',
                'animatio...    'ytick.minor.visible': 'False',
                'ytick.minor.width': '0.5',
                'ytick.right': 'True'})
key = 'backend'

    def __getitem__(self, key):
>       return self._get_context_dict().__getitem__(key)
E       KeyError: 'backend'

clean/lib/python3.11/site-packages/solara/server/patch.py:195: KeyError
```

--- a/solara/server/patch.py
+++ b/solara/server/patch.py
@@ -407,6 +407,13 @@ def patch_matplotlib():
             # same as _get
             return self[key]
 
+        def clear(self):
+            # in matplotlib .clear is effectively a no-op
+            # see https://github.com/matplotlib/matplotlib/issues/25855
+            pass
+            # in the future, we may want to clear the context dict if this is fixed
+            # self._get_context_dict().clear()
+
         def _get_context_dict(self) -> dict:
             if not self._was_initialized:
                 # since we monkey patch the class after __init__ was called

--- a/tests/unit/matplotlib_test.py
+++ b/tests/unit/matplotlib_test.py
@@ -1,8 +1,15 @@
 from matplotlib.figure import Figure
+from matplotlib import pyplot as plt
+import matplotlib
+from pathlib import Path
+
 
 import solara
 import solara.server.patch
 from solara.server import kernel
+from solara.server.app import AppScript
+
+HERE = Path(__file__).parent
 
 
 @solara.component
@@ -75,3 +82,17 @@ def test_pylab(no_kernel_context):
         cleanup()
         context_1.close()
         context_2.close()
+
+
+def test_clear_bug_in_matplotlib(no_kernel_context, kernel_context):
+    name = str(HERE / "solara_test_apps" / "single_file.py")
+    app = AppScript(name)
+    app.init()
+    try:
+        with kernel_context:
+            # see https://github.com/matplotlib/matplotlib/issues/25855
+            with plt.style.context({}, after_reset=True):
+                pass
+            matplotlib.get_backend().lower()
+    finally:
+        app.close()

